### PR TITLE
LIBCLOUD-494: Add image extension methods for create/copy 

### PR DESCRIPTION
**Pull request by cderamus**
This PR adds new extension methods to copy AMIs between Amazon regions and also adds the ability to create an image from an EBS backed instance. Unit tests and new fixtures are included.

### CODE REVIEW

File: libcloud/compute/drivers/ec2.py [hunk 1]
```diff
@@ -1566,6 +1566,105 @@ def delete_key_pair(self, key_pair):
                            namespace=NAMESPACE)
         return element == 'true'
 
+    def ex_copy_image(self, source_region, image, name=None, description=None):
+        """
+        Copy an Amazon Machine Image from one region to another.
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: The to be more clear about what is going on, the description should probably say "Copy an Amazon Machine Image from the specified source region to the current region.".

File: libcloud/compute/drivers/ec2.py [hunk 2]
```diff
@@ -1566,6 +1566,105 @@ def delete_key_pair(self, key_pair):
...
+            self.connection.request(self.path, params=params).object)
+
+        return image
+
+    def ex_create_image_from_node(self, instance, name, block_device_mapping,
+                                  no_reboot=False, description=None):
+        """
+        Create an Amazon Machine Image based off of an EBS-backed instance.
+
+        :param      instance: Instance of ``Node``
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: For consistency, the argument should be called `node`.

File: libcloud/compute/drivers/ec2.py [hunk 3]
```diff
@@ -1566,6 +1566,105 @@ def delete_key_pair(self, key_pair):
...
+
+        :param      name: The name for the new image
+        :type       name: ``str``
+
+        :param      block_device_mapping: A dictionary of the disk layout
+                                          An example of this dict is included
+                                          below.
+        :type       block_device_mapping: ``list`` of ``dict``
+
+        :param      no_reboot: Whether or not to shutdown the instance before
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: I know that the EC2 API calls the argument `NoReboot`, but I find `reboot` more clear and obvious than `no_reboot`.

File: libcloud/compute/drivers/ec2.py [hunk 4]
```diff
@@ -1566,6 +1566,105 @@ def delete_key_pair(self, key_pair):
...
+                  'NoReboot': no_reboot}
+
+        if description is not None:
+            params['Description'] = description
+
+        if not isinstance(block_device_mapping, (list, tuple)):
+            raise AttributeError(
+                'block_device_mapping not list or tuple')
+
+        for idx, mapping in enumerate(block_device_mapping):
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: Hm, this code looks familiar, have I see it before? :)

If there is a similar code repeat elsewhere in the driver, we should try to refactor it in a separate method.

File: libcloud/compute/drivers/ec2.py (hunk 1 above)
Comment (cderamus) [inline libcloud/compute/drivers/ec2.py]: Done

File: libcloud/compute/drivers/ec2.py (hunk 2 above)
Comment (cderamus) [inline libcloud/compute/drivers/ec2.py]: Roger, blind copy paste there which I should have changed.

File: libcloud/compute/drivers/ec2.py (hunk 4 above)
Comment (cderamus) [inline libcloud/compute/drivers/ec2.py]: It's a touch different than the one in create_node, but I can certainly refactor and make a utility function for it.

**Comment by Kami** [comment]
Fixed the build issue and merged patch into trunk. Thanks!

**Pull request closed without merging**